### ISSUE TITLE
form field missing a label on collection edit page

### ISSUE THREAD

**Issue by rjkati** [samvera/hyrax#6802]
### Descriptive summary

On dev.nurax.samvera.org (Hyrax 5.0.1), the second title field on the Edit Collection page does not have a label and does not meet WCAG AA requirements

### Steps to reproduce the behavior in User Interface (UI)

1. In Chrome, download the Site Improve browser extension
2. As an admin, go to https://dev.nurax.samvera.org/dashboard/collections/bc386j44x/edit?locale=en and run the Site Improve browser extension

![edit_collection_missing_label](https://github.com/samvera/hyrax/assets/32644254/ff95056a-610b-40c4-89c1-ad64bb7d5f03)
